tensorvis/tensorvisbase/src/processors/tensorfield2dmetadata.cpp

```diff
@@ -0,0 +1,132 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/tensorvisbase/processors/tensorfield2dmetadata.h>
+#include <inviwo/tensorvisbase/util/tensorutil.h>
+#include <inviwo/core/util/stringconversion.h>
+#include <inviwo/core/network/networklock.h>
+#include <inviwo/tensorvisbase/tensorvisbasemodule.h>
+#include <inviwo/core/util/constexprhash.h>
+#include <inviwo/tensorvisbase/util/attributeutil.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo TensorField2DMetaData::processorInfo_{
+    "org.inviwo.TensorField2DMetaData",  // Class identifier
+    "Tensor Field 2D Meta Data",         // Display name
+    "Data generation",                   // Category
+    CodeState::Experimental,             // Code state
+    tag::OpenTensorVis | Tag::CPU        // Tags
+};
+const ProcessorInfo TensorField2DMetaData::getProcessorInfo() const { return processorInfo_; }
+
+TensorField2DMetaData::TensorField2DMetaData()
+    : Processor()
+    , inport_("inport")
+    , outport_("outport")
+    , metaDataPropertyContainer_("metaDataPropertyContainer", "Meta data")
+    , selectAll_("selectAll", "Select all")
+    , deselectAll_("deselectAll", "Deselect all") {
+
+    addPort(inport_);
+    addPort(outport_);
+
+    addProperty(metaDataPropertyContainer_);
+    addProperty(selectAll_);
+    addProperty(deselectAll_);
+
+    inport_.onChange([this]() { invalidate(InvalidationLevel::InvalidResources); });
+
+    selectAll_.onChange([this]() {
+        auto comp = getPropertiesByType<CompositeProperty>().front()->getProperties();
+
+        for (auto prop : comp) {
+            static_cast<BoolProperty*>(prop)->set(true);
+        }
+    });
+
+    deselectAll_.onChange([this]() {
+        auto comp = getPropertiesByType<CompositeProperty>().front()->getProperties();
+
+        for (auto prop : comp) {
+            static_cast<BoolProperty*>(prop)->set(false);
+        }
+    });
+}
+
+void TensorField2DMetaData::initializeResources() {
+    if (!inport_.hasData()) return;
+
+    auto comp = getPropertiesByType<CompositeProperty>().front();
+    auto props = comp->getProperties();
+
+    NetworkLock l;
+
+    for (auto prop : props) {
+        comp->removeProperty(prop);
+    }
+
+    util::for_each_type<attributes::types2D>{}(
+        util::AddMetaDataProperties<2>{this, inport_.getData()});
+
+    /*
+    This is a bit of a hack but it'll do. What happens is that adding/removing of the default meta
+    data such as eigen values and eigen vectors is disabled by setting the respective properties to
+    readonly.
+    */
+    comp = getPropertiesByType<CompositeProperty>().front();
+    props = comp->getProperties();
+    for (size_t i{0}; i < 4; ++i) {
```

Comment:
Would it be possible to do this by identifier?

Reply:
Done.